packages/patternfly-4/react-charts/src/components/ChartUtils/chart-label.ts

```diff
@@ -1,81 +1,115 @@
-import {ChartCommonStyles, ChartThemeDefinition} from '../ChartTheme';
-import { TextSize } from 'victory-core';
+import { Helpers, TextSize } from 'victory-core';
+import { getPieOrigin } from './chart-origin';
+import { ChartCommonStyles, ChartThemeDefinition } from '../ChartTheme';
 
-interface ChartLabelPaddingXInterface {
+interface ChartBulletLabelInterface {
   dx?: number; // Horizontal shift from the x coordinate
-  chartWidth: number; // Width of chart (e.g., donut) within SVG
-  labelPosition?: 'bottom' | 'center' | 'left' | 'right' | 'top' | 'top-left'; // Position of label
+  dy?: number; // Vertical shift from the x coordinate
+  chartHeight?: number; // Width of chart within SVG
+  chartWidth?: number; // Width of chart (e.g., donut) within SVG
+  labelPosition?: 'bottom' | 'left' | 'top' | 'top-left'; // Position of label
   legendPosition?: 'bottom' | 'bottom-left' | 'right'; // Position of legend
+  svgHeight?: number; // Overall height of SVG
   svgWidth?: number; // Overall width of SVG
+  width?: number; // Chart width
 }
 
-interface ChartLabelPaddingYInterface {
-  dy?: number; // Vertical shift from the x coordinate
-  chartHeight: number; // Width of chart (e.g., donut) within SVG
-  labelPosition?: 'bottom' | 'center' | 'left' | 'right' | 'top' | 'top-left'; // Position of label
+interface ChartPieLabelInterface {
+  dx?: number; // Horizontal shift from the x coordinate
+  dy?: number; // Horizontal shift from the y coordinate
+  height: number; // Chart height
+  labelPosition?: 'bottom' | 'center' | 'right' ; // Position of label
+  legendPosition?: 'bottom' | 'right'; // Position of legend
+  padding: any; // Chart padding
+  width: number; // Chart width
 }
 
 interface ChartLabelTextSizeInterface {
   text: any; // The text to size
   theme: ChartThemeDefinition; // The theme that will be applied to the chart
 }
 
-// Returns x coordinate for label
-export const getLabelX = ({
+// Returns x coordinate for bullet labels
+export const getBulletLabelX = ({
   chartWidth,
   dx = 0,
-  labelPosition,
-  legendPosition,
-  svgWidth
-}: ChartLabelPaddingXInterface) => {
-  if (!chartWidth) {
-    return 0;
+  labelPosition
+}: ChartBulletLabelInterface) => {
+  switch (labelPosition) {
+    case 'top':
+      return chartWidth ? Math.round(chartWidth / 2) + dx : dx;
+    case 'left':
+    case 'top-left':
+    default:
+      return dx;
   }
+};
+
+// Returns y coordinate for bullet labels
+export const getBulletLabelY = ({
+  chartHeight,
+  dy = 0,
+  labelPosition
+}: ChartBulletLabelInterface) => {
   switch (labelPosition) {
-    case 'center':
-      switch (legendPosition) {
-        case 'bottom':
-          return Math.round(svgWidth / 2) + dx;
-        default:
-          return Math.round(chartWidth / 2) + dx;
-      }
     case 'bottom':
+      return chartHeight + ChartCommonStyles.label.margin + dy;
+    case 'left':
+      return chartHeight ? Math.round(chartHeight / 2) + dy : dy;
     case 'top':
```

Comment:
Unnecessary empty cases. Default case will catch anything other than the above cases.

Reply:
Removed. My intention was only to show each supported value and what it should return.